### PR TITLE
Default to OpenSSL hash function implementations

### PR DESCRIPTION
**Pull request by nbougalis**
Use OpenSSL implementations of SHA256, SHA512 and RIPEMD160 by default and add a simple (manual) benchmark unit test.

### CODE REVIEW

File: src/ripple/protocol/tests/digest_test.cpp [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+            result = d;
+        }
+
+        log << name << " results:";
+
+        auto const sum = std::accumulate(
+            results.begin(), results.end(),
+            std::chrono::nanoseconds{0});
+        {
+            auto s = std::chrono::duration_cast<std::chrono::seconds>(sum);
```
Comment (vinniefalco) [inline src/ripple/protocol/tests/digest_test.cpp]: `using namespace std::chrono` please lol

Comment (nbougalis) [inline src/ripple/protocol/tests/digest_test.cpp]: Oops!

**Comment by vinniefalco** [comment]
Suggested: `Use OpenSSL for digests`

**Comment by nbougalis** [comment]
Merged as 0e7c8ce5548705342c62941511f7877f9ca08a46

**Pull request closed without merging**